### DISCUSSION TITLE
I get 'OSError: [WinError 193] %1 is not a valid Win32 application' whenever I try to run Streamlink

### DISCUSSION THREAD

**Discussion by pathehir** [streamlink/streamlink#3605] [Q&A]
Whenever I try to run Streamlink, I get this error:

> Traceback (most recent call last):
>   File "runpy.py", line 194, in _run_module_as_main
>   File "runpy.py", line 87, in _run_code
>   File "C:\Program Files (x86)\Streamlink\bin\streamlink.exe\__main__.py", line 17, in <module>
>   File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 18, in <module>
>     from websocket import __version__ as websocket_version
>   File "C:\Program Files (x86)\Streamlink\pkgs\websocket\__init__.py", line 22, in <module>
>     from ._abnf import *
>   File "C:\Program Files (x86)\Streamlink\pkgs\websocket\_abnf.py", line 34, in <module>
>     import numpy
>   File "C:\Users\Username\AppData\Roaming\Python\Python38\site-packages\numpy\__init__.py", line 138, in <module>
>     from . import _distributor_init
>   File "C:\Users\Username\AppData\Roaming\Python\Python38\site-packages\numpy\_distributor_init.py", line 26, in <module>
>     WinDLL(os.path.abspath(filename))
>   File "ctypes\__init__.py", line 373, in __init__
> OSError: [WinError 193] %1 is not a valid Win32 application

**Comment by gravyboat** [comment]
This sort of error is usually indicative of your Python setup being broken somehow. What installation method did you use? Which version of Windows are you running? What do you get when you run streamlink with `-l debug`?

**Reply by pathehir** [reply]
Whenever I try and run `streamlink -l debug`, I get the same message:

> Traceback (most recent call last):
  File "runpy.py", line 194, in _run_module_as_main
  File "runpy.py", line 87, in _run_code
  File "C:\Program Files (x86)\Streamlink\bin\streamlink.exe\__main__.py", line 17, in <module>
  File "C:\Program Files (x86)\Streamlink\pkgs\streamlink_cli\main.py", line 18, in <module>
    from websocket import __version__ as websocket_version
  File "C:\Program Files (x86)\Streamlink\pkgs\websocket\__init__.py", line 22, in <module>
    from ._abnf import *
  File "C:\Program Files (x86)\Streamlink\pkgs\websocket\_abnf.py", line 34, in <module>
    import numpy
  File "C:\Users\Username\AppData\Roaming\Python\Python38\site-packages\numpy\__init__.py", line 138, in <module>
    from . import _distributor_init
  File "C:\Users\Username\AppData\Roaming\Python\Python38\site-packages\numpy\_distributor_init.py", line 26, in <module>
    WinDLL(os.path.abspath(filename))
  File "ctypes\__init__.py", line 373, in __init__
OSError: [WinError 193] %1 is not a valid Win32 application

**Reply by bastimeyer** [reply]
> C:\Program Files (x86)\Streamlink\pkgs\
> C:\Users\Username\AppData\Roaming\Python\Python38\site-packages\

You are mixing two different Python environments. Of course you're running into issues with that.

`C:\Program Files (x86)\Streamlink\` is the default path on Windows if you've installed Streamlink via the Windows installer and it comes in a separate Python environment, as you can read in the documentation.

`C:\Users\Username\AppData\Roaming\Python\Python38\` this is a custom Python environment installed on your system, and it looks like you've set the [`PYTHONPATH`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONPATH) / [`PYTHONHOME`](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHOME) env vars for this directory, which means by trying to run an application from a different Python env, like Streamlink installed via the installer, you are mixing up module import lookups, which leads to errors like the one above.

The `streamlink.exe` / `streamlinkw.exe` launchers that come with the Windows installer are already overriding the `PYTHONPATH` env var via `pynsist`'s launch script:
https://github.com/takluyver/pynsist/blob/2.6/nsist/commands.py#L16

**Reply by bastimeyer** [reply]
> `streamlink.exe_main.py`

Which Streamlink version is that? The 2.0.0 installer shouldn't have this file in the `bin` directory AFAIA.

**Reply by gravyboat** [reply]
Good catch @bastimeyer.
